alerting/src/main/kotlin/org/opensearch/alerting/alerts/AlertIndices.kt

```diff
@@ -273,6 +313,21 @@ class AlertIndices(
         findingHistoryIndexInitialized
     }
 
+    suspend fun createOrUpdateInitialFindingHistoryIndex(monitor: Monitor) {
+        if (monitor.dataSources?.findingsIndex == null || monitor.dataSources.findingsIndex!!.isEmpty()) {
+            return createOrUpdateInitialFindingHistoryIndex()
+        }
+        val findingsIndex = monitor.dataSources.findingsIndex!!
+        if (!clusterService.state().routingTable().hasIndex(findingsIndex)) {
+            createIndex(
+                findingsIndex,
+                findingMapping()
+            )
+        } else {
+            updateIndexMapping(findingsIndex, findingMapping(), false)
+        }
```

Comment:
Can we modify the original ``createOrUpdateInitialFindingHistoryIndex `` method to take the optional parameter ? To keep the create index logic at one place.

Reply:
in the default `createOrUpdateInitialFindingHistoryIndex` implementation they cache the state of index(i.e. whether it has been initialized or not) which differs from the implementation when a custom index name is passed. hence we can't have the logic for index creation in one place for findings and alerts indices.